{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "400cd4bd",
   "metadata": {},
   "source": [
    "# PREPROCESSING"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "87d6b13f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from sklearn.preprocessing import LabelEncoder"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cd9155ea",
   "metadata": {},
   "source": [
    "## Proteins.csv\n",
    "\n",
    "Contains protein expression data across various samples. Each column represents a sample, with identifiers indicating the condition, sex, age, and disease stage. This data is important for studying the patterns of protein expression in relation to Alzheimer's disease.​"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "id": "bb198f55",
   "metadata": {},
   "outputs": [],
   "source": [
    "proteins = pd.read_csv('csv_files/Proteins.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "id": "107093cd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TUBA1B</th>\n",
       "      <th>0.025812249</th>\n",
       "      <th>0.114868547</th>\n",
       "      <th>0.039467029</th>\n",
       "      <th>0.164589862</th>\n",
       "      <th>-0.096525527</th>\n",
       "      <th>0.019700681</th>\n",
       "      <th>-0.002591546</th>\n",
       "      <th>0.099907368</th>\n",
       "      <th>0.042163492</th>\n",
       "      <th>...</th>\n",
       "      <th>0.024021325</th>\n",
       "      <th>-0.023119985</th>\n",
       "      <th>0.261665059</th>\n",
       "      <th>0.179903164</th>\n",
       "      <th>-0.067031579</th>\n",
       "      <th>0.250224495</th>\n",
       "      <th>-0.124336088</th>\n",
       "      <th>0.08165204</th>\n",
       "      <th>0.00944257</th>\n",
       "      <th>-0.06840385</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACTG1</td>\n",
       "      <td>0.000416</td>\n",
       "      <td>-0.054160</td>\n",
       "      <td>0.100341</td>\n",
       "      <td>0.023625</td>\n",
       "      <td>-0.067120</td>\n",
       "      <td>0.048944</td>\n",
       "      <td>0.076832</td>\n",
       "      <td>0.091890</td>\n",
       "      <td>0.024801</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.162245</td>\n",
       "      <td>0.014562</td>\n",
       "      <td>0.118578</td>\n",
       "      <td>-0.028608</td>\n",
       "      <td>-0.208031</td>\n",
       "      <td>-0.037726</td>\n",
       "      <td>-0.101082</td>\n",
       "      <td>0.031508</td>\n",
       "      <td>0.085900</td>\n",
       "      <td>-0.225454</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TUBB4B</td>\n",
       "      <td>-0.088755</td>\n",
       "      <td>-0.067824</td>\n",
       "      <td>0.084958</td>\n",
       "      <td>0.072881</td>\n",
       "      <td>-0.044365</td>\n",
       "      <td>0.065666</td>\n",
       "      <td>0.017401</td>\n",
       "      <td>0.045955</td>\n",
       "      <td>0.045929</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.103476</td>\n",
       "      <td>-0.118308</td>\n",
       "      <td>0.125296</td>\n",
       "      <td>-0.100258</td>\n",
       "      <td>-0.095145</td>\n",
       "      <td>0.150440</td>\n",
       "      <td>-0.143795</td>\n",
       "      <td>0.100730</td>\n",
       "      <td>-0.005605</td>\n",
       "      <td>-0.147657</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HBB</td>\n",
       "      <td>0.157828</td>\n",
       "      <td>-0.827798</td>\n",
       "      <td>-0.050724</td>\n",
       "      <td>-0.675763</td>\n",
       "      <td>-1.104865</td>\n",
       "      <td>-0.443659</td>\n",
       "      <td>0.476736</td>\n",
       "      <td>-0.545614</td>\n",
       "      <td>-0.153280</td>\n",
       "      <td>...</td>\n",
       "      <td>0.024241</td>\n",
       "      <td>0.097263</td>\n",
       "      <td>0.415171</td>\n",
       "      <td>0.296723</td>\n",
       "      <td>0.307290</td>\n",
       "      <td>-0.099840</td>\n",
       "      <td>-0.031416</td>\n",
       "      <td>-0.273378</td>\n",
       "      <td>0.206035</td>\n",
       "      <td>0.581175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GAPDH</td>\n",
       "      <td>0.072899</td>\n",
       "      <td>-0.007606</td>\n",
       "      <td>-0.023540</td>\n",
       "      <td>0.136452</td>\n",
       "      <td>0.012224</td>\n",
       "      <td>0.051385</td>\n",
       "      <td>0.067250</td>\n",
       "      <td>0.028715</td>\n",
       "      <td>0.057526</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.160431</td>\n",
       "      <td>0.010148</td>\n",
       "      <td>0.231075</td>\n",
       "      <td>0.234622</td>\n",
       "      <td>-0.078019</td>\n",
       "      <td>0.144545</td>\n",
       "      <td>-0.031710</td>\n",
       "      <td>0.163577</td>\n",
       "      <td>0.085359</td>\n",
       "      <td>0.019764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HBA2</td>\n",
       "      <td>0.154047</td>\n",
       "      <td>-0.956292</td>\n",
       "      <td>-0.050428</td>\n",
       "      <td>-0.563935</td>\n",
       "      <td>-1.276744</td>\n",
       "      <td>-0.498043</td>\n",
       "      <td>0.382287</td>\n",
       "      <td>-0.676703</td>\n",
       "      <td>-0.030467</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.011693</td>\n",
       "      <td>0.118344</td>\n",
       "      <td>0.456283</td>\n",
       "      <td>0.305501</td>\n",
       "      <td>0.306064</td>\n",
       "      <td>-0.131145</td>\n",
       "      <td>0.003923</td>\n",
       "      <td>-0.238403</td>\n",
       "      <td>0.374431</td>\n",
       "      <td>0.461211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SPTAN1</td>\n",
       "      <td>-0.045855</td>\n",
       "      <td>-0.035560</td>\n",
       "      <td>0.022052</td>\n",
       "      <td>-0.072983</td>\n",
       "      <td>0.090556</td>\n",
       "      <td>0.019037</td>\n",
       "      <td>0.071292</td>\n",
       "      <td>-0.025271</td>\n",
       "      <td>-0.042318</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.077548</td>\n",
       "      <td>0.022194</td>\n",
       "      <td>-0.030857</td>\n",
       "      <td>-0.080586</td>\n",
       "      <td>-0.002678</td>\n",
       "      <td>0.028117</td>\n",
       "      <td>-0.066086</td>\n",
       "      <td>0.011087</td>\n",
       "      <td>0.028705</td>\n",
       "      <td>-0.076349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>DPYSL2</td>\n",
       "      <td>0.065950</td>\n",
       "      <td>0.001286</td>\n",
       "      <td>0.014256</td>\n",
       "      <td>0.189760</td>\n",
       "      <td>0.018954</td>\n",
       "      <td>0.030938</td>\n",
       "      <td>-0.091357</td>\n",
       "      <td>0.032064</td>\n",
       "      <td>0.104706</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.096669</td>\n",
       "      <td>0.024178</td>\n",
       "      <td>0.372230</td>\n",
       "      <td>0.138455</td>\n",
       "      <td>0.008448</td>\n",
       "      <td>0.204549</td>\n",
       "      <td>-0.000300</td>\n",
       "      <td>0.285529</td>\n",
       "      <td>-0.055993</td>\n",
       "      <td>-0.022201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SPTBN1</td>\n",
       "      <td>-0.015957</td>\n",
       "      <td>0.005202</td>\n",
       "      <td>-0.000139</td>\n",
       "      <td>-0.020462</td>\n",
       "      <td>0.044539</td>\n",
       "      <td>-0.052377</td>\n",
       "      <td>0.044861</td>\n",
       "      <td>-0.041211</td>\n",
       "      <td>-0.012246</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.036425</td>\n",
       "      <td>-0.017915</td>\n",
       "      <td>-0.051819</td>\n",
       "      <td>-0.051165</td>\n",
       "      <td>0.055617</td>\n",
       "      <td>0.003923</td>\n",
       "      <td>-0.010211</td>\n",
       "      <td>0.033795</td>\n",
       "      <td>-0.005548</td>\n",
       "      <td>0.042559</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CKB</td>\n",
       "      <td>-0.039686</td>\n",
       "      <td>-0.121929</td>\n",
       "      <td>0.012872</td>\n",
       "      <td>0.114658</td>\n",
       "      <td>-0.107376</td>\n",
       "      <td>-0.019236</td>\n",
       "      <td>0.091971</td>\n",
       "      <td>-0.051485</td>\n",
       "      <td>0.110854</td>\n",
       "      <td>...</td>\n",
       "      <td>0.019868</td>\n",
       "      <td>-0.024531</td>\n",
       "      <td>0.099417</td>\n",
       "      <td>0.135772</td>\n",
       "      <td>-0.123357</td>\n",
       "      <td>0.033652</td>\n",
       "      <td>0.070175</td>\n",
       "      <td>0.127060</td>\n",
       "      <td>0.047799</td>\n",
       "      <td>0.138158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>PKM</td>\n",
       "      <td>0.081944</td>\n",
       "      <td>-0.071066</td>\n",
       "      <td>0.162457</td>\n",
       "      <td>0.092396</td>\n",
       "      <td>0.079172</td>\n",
       "      <td>0.149398</td>\n",
       "      <td>0.013927</td>\n",
       "      <td>-0.004410</td>\n",
       "      <td>-0.039993</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.058542</td>\n",
       "      <td>-0.055328</td>\n",
       "      <td>0.095140</td>\n",
       "      <td>-0.130894</td>\n",
       "      <td>-0.142615</td>\n",
       "      <td>0.163489</td>\n",
       "      <td>-0.099725</td>\n",
       "      <td>-0.085582</td>\n",
       "      <td>-0.030203</td>\n",
       "      <td>-0.033967</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 420 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   TUBA1B  0.025812249  0.114868547  0.039467029  0.164589862  -0.096525527  \\\n",
       "0   ACTG1     0.000416    -0.054160     0.100341     0.023625     -0.067120   \n",
       "1  TUBB4B    -0.088755    -0.067824     0.084958     0.072881     -0.044365   \n",
       "2     HBB     0.157828    -0.827798    -0.050724    -0.675763     -1.104865   \n",
       "3   GAPDH     0.072899    -0.007606    -0.023540     0.136452      0.012224   \n",
       "4    HBA2     0.154047    -0.956292    -0.050428    -0.563935     -1.276744   \n",
       "5  SPTAN1    -0.045855    -0.035560     0.022052    -0.072983      0.090556   \n",
       "6  DPYSL2     0.065950     0.001286     0.014256     0.189760      0.018954   \n",
       "7  SPTBN1    -0.015957     0.005202    -0.000139    -0.020462      0.044539   \n",
       "8     CKB    -0.039686    -0.121929     0.012872     0.114658     -0.107376   \n",
       "9     PKM     0.081944    -0.071066     0.162457     0.092396      0.079172   \n",
       "\n",
       "   0.019700681  -0.002591546  0.099907368  0.042163492  ...  0.024021325  \\\n",
       "0     0.048944      0.076832     0.091890     0.024801  ...    -0.162245   \n",
       "1     0.065666      0.017401     0.045955     0.045929  ...    -0.103476   \n",
       "2    -0.443659      0.476736    -0.545614    -0.153280  ...     0.024241   \n",
       "3     0.051385      0.067250     0.028715     0.057526  ...    -0.160431   \n",
       "4    -0.498043      0.382287    -0.676703    -0.030467  ...    -0.011693   \n",
       "5     0.019037      0.071292    -0.025271    -0.042318  ...    -0.077548   \n",
       "6     0.030938     -0.091357     0.032064     0.104706  ...    -0.096669   \n",
       "7    -0.052377      0.044861    -0.041211    -0.012246  ...    -0.036425   \n",
       "8    -0.019236      0.091971    -0.051485     0.110854  ...     0.019868   \n",
       "9     0.149398      0.013927    -0.004410    -0.039993  ...    -0.058542   \n",
       "\n",
       "   -0.023119985  0.261665059  0.179903164  -0.067031579  0.250224495  \\\n",
       "0      0.014562     0.118578    -0.028608     -0.208031    -0.037726   \n",
       "1     -0.118308     0.125296    -0.100258     -0.095145     0.150440   \n",
       "2      0.097263     0.415171     0.296723      0.307290    -0.099840   \n",
       "3      0.010148     0.231075     0.234622     -0.078019     0.144545   \n",
       "4      0.118344     0.456283     0.305501      0.306064    -0.131145   \n",
       "5      0.022194    -0.030857    -0.080586     -0.002678     0.028117   \n",
       "6      0.024178     0.372230     0.138455      0.008448     0.204549   \n",
       "7     -0.017915    -0.051819    -0.051165      0.055617     0.003923   \n",
       "8     -0.024531     0.099417     0.135772     -0.123357     0.033652   \n",
       "9     -0.055328     0.095140    -0.130894     -0.142615     0.163489   \n",
       "\n",
       "   -0.124336088  0.08165204  0.00944257  -0.06840385  \n",
       "0     -0.101082    0.031508    0.085900    -0.225454  \n",
       "1     -0.143795    0.100730   -0.005605    -0.147657  \n",
       "2     -0.031416   -0.273378    0.206035     0.581175  \n",
       "3     -0.031710    0.163577    0.085359     0.019764  \n",
       "4      0.003923   -0.238403    0.374431     0.461211  \n",
       "5     -0.066086    0.011087    0.028705    -0.076349  \n",
       "6     -0.000300    0.285529   -0.055993    -0.022201  \n",
       "7     -0.010211    0.033795   -0.005548     0.042559  \n",
       "8      0.070175    0.127060    0.047799     0.138158  \n",
       "9     -0.099725   -0.085582   -0.030203    -0.033967  \n",
       "\n",
       "[10 rows x 420 columns]"
      ]
     },
     "execution_count": 132,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteins.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "420736d2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>3321</th>\n",
       "      <th>3322</th>\n",
       "      <th>3323</th>\n",
       "      <th>3324</th>\n",
       "      <th>3325</th>\n",
       "      <th>3326</th>\n",
       "      <th>3327</th>\n",
       "      <th>3328</th>\n",
       "      <th>3329</th>\n",
       "      <th>3330</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TUBA1B</th>\n",
       "      <td>ACTG1</td>\n",
       "      <td>TUBB4B</td>\n",
       "      <td>HBB</td>\n",
       "      <td>GAPDH</td>\n",
       "      <td>HBA2</td>\n",
       "      <td>SPTAN1</td>\n",
       "      <td>DPYSL2</td>\n",
       "      <td>SPTBN1</td>\n",
       "      <td>CKB</td>\n",
       "      <td>PKM</td>\n",
       "      <td>...</td>\n",
       "      <td>AGFG2</td>\n",
       "      <td>PTRH2</td>\n",
       "      <td>TMED5</td>\n",
       "      <td>PODXL2</td>\n",
       "      <td>CUL4A</td>\n",
       "      <td>REEP6</td>\n",
       "      <td>NPC1</td>\n",
       "      <td>DSCR3</td>\n",
       "      <td>HAX1</td>\n",
       "      <td>VKORC1L1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.025812249</th>\n",
       "      <td>0.000416</td>\n",
       "      <td>-0.088755</td>\n",
       "      <td>0.157828</td>\n",
       "      <td>0.072899</td>\n",
       "      <td>0.154047</td>\n",
       "      <td>-0.045855</td>\n",
       "      <td>0.06595</td>\n",
       "      <td>-0.015957</td>\n",
       "      <td>-0.039686</td>\n",
       "      <td>0.081944</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.086443</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.633261</td>\n",
       "      <td>0.159727</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.429527</td>\n",
       "      <td>-0.001607</td>\n",
       "      <td>0.590137</td>\n",
       "      <td>-0.116056</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.114868547</th>\n",
       "      <td>-0.05416</td>\n",
       "      <td>-0.067824</td>\n",
       "      <td>-0.827798</td>\n",
       "      <td>-0.007606</td>\n",
       "      <td>-0.956292</td>\n",
       "      <td>-0.03556</td>\n",
       "      <td>0.001286</td>\n",
       "      <td>0.005202</td>\n",
       "      <td>-0.121929</td>\n",
       "      <td>-0.071066</td>\n",
       "      <td>...</td>\n",
       "      <td>0.03406</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.087358</td>\n",
       "      <td>-0.002128</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.011036</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.171433</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.039467029</th>\n",
       "      <td>0.100341</td>\n",
       "      <td>0.084958</td>\n",
       "      <td>-0.050724</td>\n",
       "      <td>-0.02354</td>\n",
       "      <td>-0.050428</td>\n",
       "      <td>0.022052</td>\n",
       "      <td>0.014256</td>\n",
       "      <td>-0.000139</td>\n",
       "      <td>0.012872</td>\n",
       "      <td>0.162457</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.410899</td>\n",
       "      <td>-0.177223</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.179217</td>\n",
       "      <td>-0.527957</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.790274</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.164589862</th>\n",
       "      <td>0.023625</td>\n",
       "      <td>0.072881</td>\n",
       "      <td>-0.675763</td>\n",
       "      <td>0.136452</td>\n",
       "      <td>-0.563935</td>\n",
       "      <td>-0.072983</td>\n",
       "      <td>0.18976</td>\n",
       "      <td>-0.020462</td>\n",
       "      <td>0.114658</td>\n",
       "      <td>0.092396</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.33279</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.213148</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.128936</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>-0.096525527</th>\n",
       "      <td>-0.06712</td>\n",
       "      <td>-0.044365</td>\n",
       "      <td>-1.104865</td>\n",
       "      <td>0.012224</td>\n",
       "      <td>-1.276744</td>\n",
       "      <td>0.090556</td>\n",
       "      <td>0.018954</td>\n",
       "      <td>0.044539</td>\n",
       "      <td>-0.107376</td>\n",
       "      <td>0.079172</td>\n",
       "      <td>...</td>\n",
       "      <td>0.591717</td>\n",
       "      <td>1.484447</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.261398</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.23453</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.623527</td>\n",
       "      <td>0.02367</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.019700681</th>\n",
       "      <td>0.048944</td>\n",
       "      <td>0.065666</td>\n",
       "      <td>-0.443659</td>\n",
       "      <td>0.051385</td>\n",
       "      <td>-0.498043</td>\n",
       "      <td>0.019037</td>\n",
       "      <td>0.030938</td>\n",
       "      <td>-0.052377</td>\n",
       "      <td>-0.019236</td>\n",
       "      <td>0.149398</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.236246</td>\n",
       "      <td>0.551475</td>\n",
       "      <td>1.272439</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.452177</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>-0.002591546</th>\n",
       "      <td>0.076832</td>\n",
       "      <td>0.017401</td>\n",
       "      <td>0.476736</td>\n",
       "      <td>0.06725</td>\n",
       "      <td>0.382287</td>\n",
       "      <td>0.071292</td>\n",
       "      <td>-0.091357</td>\n",
       "      <td>0.044861</td>\n",
       "      <td>0.091971</td>\n",
       "      <td>0.013927</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.017237</td>\n",
       "      <td>0.137227</td>\n",
       "      <td>-0.819154</td>\n",
       "      <td>0.244657</td>\n",
       "      <td>-2.164349</td>\n",
       "      <td>-0.235973</td>\n",
       "      <td>-0.569174</td>\n",
       "      <td>0.188316</td>\n",
       "      <td>-0.407177</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.099907368</th>\n",
       "      <td>0.09189</td>\n",
       "      <td>0.045955</td>\n",
       "      <td>-0.545614</td>\n",
       "      <td>0.028715</td>\n",
       "      <td>-0.676703</td>\n",
       "      <td>-0.025271</td>\n",
       "      <td>0.032064</td>\n",
       "      <td>-0.041211</td>\n",
       "      <td>-0.051485</td>\n",
       "      <td>-0.00441</td>\n",
       "      <td>...</td>\n",
       "      <td>0.060379</td>\n",
       "      <td>-0.056896</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.28865</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.98305</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.312086</td>\n",
       "      <td>0.697252</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.042163492</th>\n",
       "      <td>0.024801</td>\n",
       "      <td>0.045929</td>\n",
       "      <td>-0.15328</td>\n",
       "      <td>0.057526</td>\n",
       "      <td>-0.030467</td>\n",
       "      <td>-0.042318</td>\n",
       "      <td>0.104706</td>\n",
       "      <td>-0.012246</td>\n",
       "      <td>0.110854</td>\n",
       "      <td>-0.039993</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.198928</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.137645</td>\n",
       "      <td>0.574422</td>\n",
       "      <td>0.237895</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 3331 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  0         1         2         3         4         5     \\\n",
       "TUBA1B           ACTG1    TUBB4B       HBB     GAPDH      HBA2    SPTAN1   \n",
       "0.025812249   0.000416 -0.088755  0.157828  0.072899  0.154047 -0.045855   \n",
       "0.114868547   -0.05416 -0.067824 -0.827798 -0.007606 -0.956292  -0.03556   \n",
       "0.039467029   0.100341  0.084958 -0.050724  -0.02354 -0.050428  0.022052   \n",
       "0.164589862   0.023625  0.072881 -0.675763  0.136452 -0.563935 -0.072983   \n",
       "-0.096525527  -0.06712 -0.044365 -1.104865  0.012224 -1.276744  0.090556   \n",
       "0.019700681   0.048944  0.065666 -0.443659  0.051385 -0.498043  0.019037   \n",
       "-0.002591546  0.076832  0.017401  0.476736   0.06725  0.382287  0.071292   \n",
       "0.099907368    0.09189  0.045955 -0.545614  0.028715 -0.676703 -0.025271   \n",
       "0.042163492   0.024801  0.045929  -0.15328  0.057526 -0.030467 -0.042318   \n",
       "\n",
       "                  6         7         8         9     ...      3321      3322  \\\n",
       "TUBA1B          DPYSL2    SPTBN1       CKB       PKM  ...     AGFG2     PTRH2   \n",
       "0.025812249    0.06595 -0.015957 -0.039686  0.081944  ... -0.086443       NaN   \n",
       "0.114868547   0.001286  0.005202 -0.121929 -0.071066  ...   0.03406       NaN   \n",
       "0.039467029   0.014256 -0.000139  0.012872  0.162457  ...       NaN -0.410899   \n",
       "0.164589862    0.18976 -0.020462  0.114658  0.092396  ...  -0.33279       NaN   \n",
       "-0.096525527  0.018954  0.044539 -0.107376  0.079172  ...  0.591717  1.484447   \n",
       "0.019700681   0.030938 -0.052377 -0.019236  0.149398  ...       NaN -0.236246   \n",
       "-0.002591546 -0.091357  0.044861  0.091971  0.013927  ...       NaN  0.017237   \n",
       "0.099907368   0.032064 -0.041211 -0.051485  -0.00441  ...  0.060379 -0.056896   \n",
       "0.042163492   0.104706 -0.012246  0.110854 -0.039993  ... -0.198928       NaN   \n",
       "\n",
       "                  3323      3324      3325      3326      3327      3328  \\\n",
       "TUBA1B           TMED5    PODXL2     CUL4A     REEP6      NPC1     DSCR3   \n",
       "0.025812249  -0.633261  0.159727       NaN  0.429527 -0.001607  0.590137   \n",
       "0.114868547  -0.087358 -0.002128       NaN -0.011036       NaN       NaN   \n",
       "0.039467029  -0.177223       NaN -0.179217 -0.527957       NaN -0.790274   \n",
       "0.164589862        NaN       NaN       NaN  0.213148       NaN -0.128936   \n",
       "-0.096525527       NaN  0.261398       NaN   0.23453       NaN  0.623527   \n",
       "0.019700681   0.551475  1.272439       NaN       NaN       NaN  0.452177   \n",
       "-0.002591546  0.137227 -0.819154  0.244657 -2.164349 -0.235973 -0.569174   \n",
       "0.099907368        NaN  -0.28865       NaN  -0.98305       NaN -0.312086   \n",
       "0.042163492        NaN  0.137645  0.574422  0.237895       NaN       NaN   \n",
       "\n",
       "                  3329      3330  \n",
       "TUBA1B            HAX1  VKORC1L1  \n",
       "0.025812249  -0.116056       NaN  \n",
       "0.114868547  -0.171433       NaN  \n",
       "0.039467029        NaN       NaN  \n",
       "0.164589862        NaN       NaN  \n",
       "-0.096525527   0.02367       NaN  \n",
       "0.019700681        NaN       NaN  \n",
       "-0.002591546  0.188316 -0.407177  \n",
       "0.099907368   0.697252       NaN  \n",
       "0.042163492        NaN       NaN  \n",
       "\n",
       "[10 rows x 3331 columns]"
      ]
     },
     "execution_count": 133,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Transpose the DataFrame\n",
    "proteins_transpose = proteins.transpose()\n",
    "proteins_transpose.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "id": "e2b5b7a9",
   "metadata": {},
   "outputs": [],
   "source": [
    "proteins_transpose.columns = proteins_transpose.iloc[0]\n",
    "proteins_transpose = proteins_transpose.drop(proteins_transpose.index[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "id": "0cd49780",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>TUBA1B</th>\n",
       "      <th>ACTG1</th>\n",
       "      <th>TUBB4B</th>\n",
       "      <th>HBB</th>\n",
       "      <th>GAPDH</th>\n",
       "      <th>HBA2</th>\n",
       "      <th>SPTAN1</th>\n",
       "      <th>DPYSL2</th>\n",
       "      <th>SPTBN1</th>\n",
       "      <th>CKB</th>\n",
       "      <th>PKM</th>\n",
       "      <th>...</th>\n",
       "      <th>AGFG2</th>\n",
       "      <th>PTRH2</th>\n",
       "      <th>TMED5</th>\n",
       "      <th>PODXL2</th>\n",
       "      <th>CUL4A</th>\n",
       "      <th>REEP6</th>\n",
       "      <th>NPC1</th>\n",
       "      <th>DSCR3</th>\n",
       "      <th>HAX1</th>\n",
       "      <th>VKORC1L1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0.025812249</th>\n",
       "      <td>0.000416</td>\n",
       "      <td>-0.088755</td>\n",
       "      <td>0.157828</td>\n",
       "      <td>0.072899</td>\n",
       "      <td>0.154047</td>\n",
       "      <td>-0.045855</td>\n",
       "      <td>0.06595</td>\n",
       "      <td>-0.015957</td>\n",
       "      <td>-0.039686</td>\n",
       "      <td>0.081944</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.086443</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.633261</td>\n",
       "      <td>0.159727</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.429527</td>\n",
       "      <td>-0.001607</td>\n",
       "      <td>0.590137</td>\n",
       "      <td>-0.116056</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.114868547</th>\n",
       "      <td>-0.05416</td>\n",
       "      <td>-0.067824</td>\n",
       "      <td>-0.827798</td>\n",
       "      <td>-0.007606</td>\n",
       "      <td>-0.956292</td>\n",
       "      <td>-0.03556</td>\n",
       "      <td>0.001286</td>\n",
       "      <td>0.005202</td>\n",
       "      <td>-0.121929</td>\n",
       "      <td>-0.071066</td>\n",
       "      <td>...</td>\n",
       "      <td>0.03406</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.087358</td>\n",
       "      <td>-0.002128</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.011036</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.171433</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.039467029</th>\n",
       "      <td>0.100341</td>\n",
       "      <td>0.084958</td>\n",
       "      <td>-0.050724</td>\n",
       "      <td>-0.02354</td>\n",
       "      <td>-0.050428</td>\n",
       "      <td>0.022052</td>\n",
       "      <td>0.014256</td>\n",
       "      <td>-0.000139</td>\n",
       "      <td>0.012872</td>\n",
       "      <td>0.162457</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.410899</td>\n",
       "      <td>-0.177223</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.179217</td>\n",
       "      <td>-0.527957</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.790274</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.164589862</th>\n",
       "      <td>0.023625</td>\n",
       "      <td>0.072881</td>\n",
       "      <td>-0.675763</td>\n",
       "      <td>0.136452</td>\n",
       "      <td>-0.563935</td>\n",
       "      <td>-0.072983</td>\n",
       "      <td>0.18976</td>\n",
       "      <td>-0.020462</td>\n",
       "      <td>0.114658</td>\n",
       "      <td>0.092396</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.33279</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.213148</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.128936</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>-0.096525527</th>\n",
       "      <td>-0.06712</td>\n",
       "      <td>-0.044365</td>\n",
       "      <td>-1.104865</td>\n",
       "      <td>0.012224</td>\n",
       "      <td>-1.276744</td>\n",
       "      <td>0.090556</td>\n",
       "      <td>0.018954</td>\n",
       "      <td>0.044539</td>\n",
       "      <td>-0.107376</td>\n",
       "      <td>0.079172</td>\n",
       "      <td>...</td>\n",
       "      <td>0.591717</td>\n",
       "      <td>1.484447</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.261398</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.23453</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.623527</td>\n",
       "      <td>0.02367</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.019700681</th>\n",
       "      <td>0.048944</td>\n",
       "      <td>0.065666</td>\n",
       "      <td>-0.443659</td>\n",
       "      <td>0.051385</td>\n",
       "      <td>-0.498043</td>\n",
       "      <td>0.019037</td>\n",
       "      <td>0.030938</td>\n",
       "      <td>-0.052377</td>\n",
       "      <td>-0.019236</td>\n",
       "      <td>0.149398</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.236246</td>\n",
       "      <td>0.551475</td>\n",
       "      <td>1.272439</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.452177</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>-0.002591546</th>\n",
       "      <td>0.076832</td>\n",
       "      <td>0.017401</td>\n",
       "      <td>0.476736</td>\n",
       "      <td>0.06725</td>\n",
       "      <td>0.382287</td>\n",
       "      <td>0.071292</td>\n",
       "      <td>-0.091357</td>\n",
       "      <td>0.044861</td>\n",
       "      <td>0.091971</td>\n",
       "      <td>0.013927</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.017237</td>\n",
       "      <td>0.137227</td>\n",
       "      <td>-0.819154</td>\n",
       "      <td>0.244657</td>\n",
       "      <td>-2.164349</td>\n",
       "      <td>-0.235973</td>\n",
       "      <td>-0.569174</td>\n",
       "      <td>0.188316</td>\n",
       "      <td>-0.407177</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.099907368</th>\n",
       "      <td>0.09189</td>\n",
       "      <td>0.045955</td>\n",
       "      <td>-0.545614</td>\n",
       "      <td>0.028715</td>\n",
       "      <td>-0.676703</td>\n",
       "      <td>-0.025271</td>\n",
       "      <td>0.032064</td>\n",
       "      <td>-0.041211</td>\n",
       "      <td>-0.051485</td>\n",
       "      <td>-0.00441</td>\n",
       "      <td>...</td>\n",
       "      <td>0.060379</td>\n",
       "      <td>-0.056896</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.28865</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.98305</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.312086</td>\n",
       "      <td>0.697252</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.042163492</th>\n",
       "      <td>0.024801</td>\n",
       "      <td>0.045929</td>\n",
       "      <td>-0.15328</td>\n",
       "      <td>0.057526</td>\n",
       "      <td>-0.030467</td>\n",
       "      <td>-0.042318</td>\n",
       "      <td>0.104706</td>\n",
       "      <td>-0.012246</td>\n",
       "      <td>0.110854</td>\n",
       "      <td>-0.039993</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.198928</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.137645</td>\n",
       "      <td>0.574422</td>\n",
       "      <td>0.237895</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.140697359</th>\n",
       "      <td>0.08091</td>\n",
       "      <td>0.097695</td>\n",
       "      <td>-0.226023</td>\n",
       "      <td>-0.106895</td>\n",
       "      <td>-0.316395</td>\n",
       "      <td>-0.073514</td>\n",
       "      <td>0.016917</td>\n",
       "      <td>-0.091123</td>\n",
       "      <td>-0.000417</td>\n",
       "      <td>0.067669</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.127688</td>\n",
       "      <td>-0.173111</td>\n",
       "      <td>0.234793</td>\n",
       "      <td>-0.018004</td>\n",
       "      <td>-0.339277</td>\n",
       "      <td>-0.96954</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.080315</td>\n",
       "      <td>0.193889</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 3331 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "TUBA1B           ACTG1    TUBB4B       HBB     GAPDH      HBA2    SPTAN1  \\\n",
       "0.025812249   0.000416 -0.088755  0.157828  0.072899  0.154047 -0.045855   \n",
       "0.114868547   -0.05416 -0.067824 -0.827798 -0.007606 -0.956292  -0.03556   \n",
       "0.039467029   0.100341  0.084958 -0.050724  -0.02354 -0.050428  0.022052   \n",
       "0.164589862   0.023625  0.072881 -0.675763  0.136452 -0.563935 -0.072983   \n",
       "-0.096525527  -0.06712 -0.044365 -1.104865  0.012224 -1.276744  0.090556   \n",
       "0.019700681   0.048944  0.065666 -0.443659  0.051385 -0.498043  0.019037   \n",
       "-0.002591546  0.076832  0.017401  0.476736   0.06725  0.382287  0.071292   \n",
       "0.099907368    0.09189  0.045955 -0.545614  0.028715 -0.676703 -0.025271   \n",
       "0.042163492   0.024801  0.045929  -0.15328  0.057526 -0.030467 -0.042318   \n",
       "0.140697359    0.08091  0.097695 -0.226023 -0.106895 -0.316395 -0.073514   \n",
       "\n",
       "TUBA1B          DPYSL2    SPTBN1       CKB       PKM  ...     AGFG2     PTRH2  \\\n",
       "0.025812249    0.06595 -0.015957 -0.039686  0.081944  ... -0.086443       NaN   \n",
       "0.114868547   0.001286  0.005202 -0.121929 -0.071066  ...   0.03406       NaN   \n",
       "0.039467029   0.014256 -0.000139  0.012872  0.162457  ...       NaN -0.410899   \n",
       "0.164589862    0.18976 -0.020462  0.114658  0.092396  ...  -0.33279       NaN   \n",
       "-0.096525527  0.018954  0.044539 -0.107376  0.079172  ...  0.591717  1.484447   \n",
       "0.019700681   0.030938 -0.052377 -0.019236  0.149398  ...       NaN -0.236246   \n",
       "-0.002591546 -0.091357  0.044861  0.091971  0.013927  ...       NaN  0.017237   \n",
       "0.099907368   0.032064 -0.041211 -0.051485  -0.00441  ...  0.060379 -0.056896   \n",
       "0.042163492   0.104706 -0.012246  0.110854 -0.039993  ... -0.198928       NaN   \n",
       "0.140697359   0.016917 -0.091123 -0.000417  0.067669  ... -0.127688 -0.173111   \n",
       "\n",
       "TUBA1B           TMED5    PODXL2     CUL4A     REEP6      NPC1     DSCR3  \\\n",
       "0.025812249  -0.633261  0.159727       NaN  0.429527 -0.001607  0.590137   \n",
       "0.114868547  -0.087358 -0.002128       NaN -0.011036       NaN       NaN   \n",
       "0.039467029  -0.177223       NaN -0.179217 -0.527957       NaN -0.790274   \n",
       "0.164589862        NaN       NaN       NaN  0.213148       NaN -0.128936   \n",
       "-0.096525527       NaN  0.261398       NaN   0.23453       NaN  0.623527   \n",
       "0.019700681   0.551475  1.272439       NaN       NaN       NaN  0.452177   \n",
       "-0.002591546  0.137227 -0.819154  0.244657 -2.164349 -0.235973 -0.569174   \n",
       "0.099907368        NaN  -0.28865       NaN  -0.98305       NaN -0.312086   \n",
       "0.042163492        NaN  0.137645  0.574422  0.237895       NaN       NaN   \n",
       "0.140697359   0.234793 -0.018004 -0.339277  -0.96954       NaN -0.080315   \n",
       "\n",
       "TUBA1B            HAX1  VKORC1L1  \n",
       "0.025812249  -0.116056       NaN  \n",
       "0.114868547  -0.171433       NaN  \n",
       "0.039467029        NaN       NaN  \n",
       "0.164589862        NaN       NaN  \n",
       "-0.096525527   0.02367       NaN  \n",
       "0.019700681        NaN       NaN  \n",
       "-0.002591546  0.188316 -0.407177  \n",
       "0.099907368   0.697252       NaN  \n",
       "0.042163492        NaN       NaN  \n",
       "0.140697359   0.193889       NaN  \n",
       "\n",
       "[10 rows x 3331 columns]"
      ]
     },
     "execution_count": 135,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteins_transpose.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "id": "c4d44784",
   "metadata": {},
   "outputs": [],
   "source": [
    "# checking for missing values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "id": "9d7776b4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "TUBA1B\n",
       "ACTG1       False\n",
       "TUBB4B      False\n",
       "HBB         False\n",
       "GAPDH       False\n",
       "HBA2        False\n",
       "            ...  \n",
       "REEP6        True\n",
       "NPC1         True\n",
       "DSCR3        True\n",
       "HAX1         True\n",
       "VKORC1L1     True\n",
       "Length: 3331, dtype: bool"
      ]
     },
     "execution_count": 137,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteins_missing_val = proteins_transpose.isnull().any()\n",
    "proteins_missing_val"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "id": "910801b3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Reset index for proper handling\n",
    "proteins_transpose.reset_index(drop=True, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "id": "f8d30228",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>TUBA1B</th>\n",
       "      <th>ACTG1</th>\n",
       "      <th>TUBB4B</th>\n",
       "      <th>HBB</th>\n",
       "      <th>GAPDH</th>\n",
       "      <th>HBA2</th>\n",
       "      <th>SPTAN1</th>\n",
       "      <th>DPYSL2</th>\n",
       "      <th>SPTBN1</th>\n",
       "      <th>CKB</th>\n",
       "      <th>PKM</th>\n",
       "      <th>...</th>\n",
       "      <th>AGFG2</th>\n",
       "      <th>PTRH2</th>\n",
       "      <th>TMED5</th>\n",
       "      <th>PODXL2</th>\n",
       "      <th>CUL4A</th>\n",
       "      <th>REEP6</th>\n",
       "      <th>NPC1</th>\n",
       "      <th>DSCR3</th>\n",
       "      <th>HAX1</th>\n",
       "      <th>VKORC1L1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.000416</td>\n",
       "      <td>-0.088755</td>\n",
       "      <td>0.157828</td>\n",
       "      <td>0.072899</td>\n",
       "      <td>0.154047</td>\n",
       "      <td>-0.045855</td>\n",
       "      <td>0.06595</td>\n",
       "      <td>-0.015957</td>\n",
       "      <td>-0.039686</td>\n",
       "      <td>0.081944</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.086443</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.633261</td>\n",
       "      <td>0.159727</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.429527</td>\n",
       "      <td>-0.001607</td>\n",
       "      <td>0.590137</td>\n",
       "      <td>-0.116056</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>-0.05416</td>\n",
       "      <td>-0.067824</td>\n",
       "      <td>-0.827798</td>\n",
       "      <td>-0.007606</td>\n",
       "      <td>-0.956292</td>\n",
       "      <td>-0.03556</td>\n",
       "      <td>0.001286</td>\n",
       "      <td>0.005202</td>\n",
       "      <td>-0.121929</td>\n",
       "      <td>-0.071066</td>\n",
       "      <td>...</td>\n",
       "      <td>0.03406</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.087358</td>\n",
       "      <td>-0.002128</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.011036</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.171433</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.100341</td>\n",
       "      <td>0.084958</td>\n",
       "      <td>-0.050724</td>\n",
       "      <td>-0.02354</td>\n",
       "      <td>-0.050428</td>\n",
       "      <td>0.022052</td>\n",
       "      <td>0.014256</td>\n",
       "      <td>-0.000139</td>\n",
       "      <td>0.012872</td>\n",
       "      <td>0.162457</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.410899</td>\n",
       "      <td>-0.177223</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.179217</td>\n",
       "      <td>-0.527957</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.790274</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.023625</td>\n",
       "      <td>0.072881</td>\n",
       "      <td>-0.675763</td>\n",
       "      <td>0.136452</td>\n",
       "      <td>-0.563935</td>\n",
       "      <td>-0.072983</td>\n",
       "      <td>0.18976</td>\n",
       "      <td>-0.020462</td>\n",
       "      <td>0.114658</td>\n",
       "      <td>0.092396</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.33279</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.213148</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.128936</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>-0.06712</td>\n",
       "      <td>-0.044365</td>\n",
       "      <td>-1.104865</td>\n",
       "      <td>0.012224</td>\n",
       "      <td>-1.276744</td>\n",
       "      <td>0.090556</td>\n",
       "      <td>0.018954</td>\n",
       "      <td>0.044539</td>\n",
       "      <td>-0.107376</td>\n",
       "      <td>0.079172</td>\n",
       "      <td>...</td>\n",
       "      <td>0.591717</td>\n",
       "      <td>1.484447</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.261398</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.234530</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.623527</td>\n",
       "      <td>0.02367</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0.048944</td>\n",
       "      <td>0.065666</td>\n",
       "      <td>-0.443659</td>\n",
       "      <td>0.051385</td>\n",
       "      <td>-0.498043</td>\n",
       "      <td>0.019037</td>\n",
       "      <td>0.030938</td>\n",
       "      <td>-0.052377</td>\n",
       "      <td>-0.019236</td>\n",
       "      <td>0.149398</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.236246</td>\n",
       "      <td>0.551475</td>\n",
       "      <td>1.272439</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.452177</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>0.076832</td>\n",
       "      <td>0.017401</td>\n",
       "      <td>0.476736</td>\n",
       "      <td>0.06725</td>\n",
       "      <td>0.382287</td>\n",
       "      <td>0.071292</td>\n",
       "      <td>-0.091357</td>\n",
       "      <td>0.044861</td>\n",
       "      <td>0.091971</td>\n",
       "      <td>0.013927</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.017237</td>\n",
       "      <td>0.137227</td>\n",
       "      <td>-0.819154</td>\n",
       "      <td>0.244657</td>\n",
       "      <td>-2.164349</td>\n",
       "      <td>-0.235973</td>\n",
       "      <td>-0.569174</td>\n",
       "      <td>0.188316</td>\n",
       "      <td>-0.407177</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>0.09189</td>\n",
       "      <td>0.045955</td>\n",
       "      <td>-0.545614</td>\n",
       "      <td>0.028715</td>\n",
       "      <td>-0.676703</td>\n",
       "      <td>-0.025271</td>\n",
       "      <td>0.032064</td>\n",
       "      <td>-0.041211</td>\n",
       "      <td>-0.051485</td>\n",
       "      <td>-0.00441</td>\n",
       "      <td>...</td>\n",
       "      <td>0.060379</td>\n",
       "      <td>-0.056896</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.28865</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.983050</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.312086</td>\n",
       "      <td>0.697252</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>0.024801</td>\n",
       "      <td>0.045929</td>\n",
       "      <td>-0.15328</td>\n",
       "      <td>0.057526</td>\n",
       "      <td>-0.030467</td>\n",
       "      <td>-0.042318</td>\n",
       "      <td>0.104706</td>\n",
       "      <td>-0.012246</td>\n",
       "      <td>0.110854</td>\n",
       "      <td>-0.039993</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.198928</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.137645</td>\n",
       "      <td>0.574422</td>\n",
       "      <td>0.237895</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>0.08091</td>\n",
       "      <td>0.097695</td>\n",
       "      <td>-0.226023</td>\n",
       "      <td>-0.106895</td>\n",
       "      <td>-0.316395</td>\n",
       "      <td>-0.073514</td>\n",
       "      <td>0.016917</td>\n",
       "      <td>-0.091123</td>\n",
       "      <td>-0.000417</td>\n",
       "      <td>0.067669</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.127688</td>\n",
       "      <td>-0.173111</td>\n",
       "      <td>0.234793</td>\n",
       "      <td>-0.018004</td>\n",
       "      <td>-0.339277</td>\n",
       "      <td>-0.969540</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.080315</td>\n",
       "      <td>0.193889</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 3331 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "TUBA1B     ACTG1    TUBB4B       HBB     GAPDH      HBA2    SPTAN1    DPYSL2  \\\n",
       "0       0.000416 -0.088755  0.157828  0.072899  0.154047 -0.045855   0.06595   \n",
       "1       -0.05416 -0.067824 -0.827798 -0.007606 -0.956292  -0.03556  0.001286   \n",
       "2       0.100341  0.084958 -0.050724  -0.02354 -0.050428  0.022052  0.014256   \n",
       "3       0.023625  0.072881 -0.675763  0.136452 -0.563935 -0.072983   0.18976   \n",
       "4       -0.06712 -0.044365 -1.104865  0.012224 -1.276744  0.090556  0.018954   \n",
       "5       0.048944  0.065666 -0.443659  0.051385 -0.498043  0.019037  0.030938   \n",
       "6       0.076832  0.017401  0.476736   0.06725  0.382287  0.071292 -0.091357   \n",
       "7        0.09189  0.045955 -0.545614  0.028715 -0.676703 -0.025271  0.032064   \n",
       "8       0.024801  0.045929  -0.15328  0.057526 -0.030467 -0.042318  0.104706   \n",
       "9        0.08091  0.097695 -0.226023 -0.106895 -0.316395 -0.073514  0.016917   \n",
       "\n",
       "TUBA1B    SPTBN1       CKB       PKM  ...     AGFG2     PTRH2     TMED5  \\\n",
       "0      -0.015957 -0.039686  0.081944  ... -0.086443       NaN -0.633261   \n",
       "1       0.005202 -0.121929 -0.071066  ...   0.03406       NaN -0.087358   \n",
       "2      -0.000139  0.012872  0.162457  ...       NaN -0.410899 -0.177223   \n",
       "3      -0.020462  0.114658  0.092396  ...  -0.33279       NaN       NaN   \n",
       "4       0.044539 -0.107376  0.079172  ...  0.591717  1.484447       NaN   \n",
       "5      -0.052377 -0.019236  0.149398  ...       NaN -0.236246  0.551475   \n",
       "6       0.044861  0.091971  0.013927  ...       NaN  0.017237  0.137227   \n",
       "7      -0.041211 -0.051485  -0.00441  ...  0.060379 -0.056896       NaN   \n",
       "8      -0.012246  0.110854 -0.039993  ... -0.198928       NaN       NaN   \n",
       "9      -0.091123 -0.000417  0.067669  ... -0.127688 -0.173111  0.234793   \n",
       "\n",
       "TUBA1B    PODXL2     CUL4A     REEP6      NPC1     DSCR3      HAX1  VKORC1L1  \n",
       "0       0.159727       NaN  0.429527 -0.001607  0.590137 -0.116056       NaN  \n",
       "1      -0.002128       NaN -0.011036       NaN       NaN -0.171433       NaN  \n",
       "2            NaN -0.179217 -0.527957       NaN -0.790274       NaN       NaN  \n",
       "3            NaN       NaN  0.213148       NaN -0.128936       NaN       NaN  \n",
       "4       0.261398       NaN  0.234530       NaN  0.623527   0.02367       NaN  \n",
       "5       1.272439       NaN       NaN       NaN  0.452177       NaN       NaN  \n",
       "6      -0.819154  0.244657 -2.164349 -0.235973 -0.569174  0.188316 -0.407177  \n",
       "7       -0.28865       NaN -0.983050       NaN -0.312086  0.697252       NaN  \n",
       "8       0.137645  0.574422  0.237895       NaN       NaN       NaN       NaN  \n",
       "9      -0.018004 -0.339277 -0.969540       NaN -0.080315  0.193889       NaN  \n",
       "\n",
       "[10 rows x 3331 columns]"
      ]
     },
     "execution_count": 152,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteins_transpose.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "id": "4c74f5e6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "TUBA1B\n",
       "NPC1   -0.041679\n",
       "dtype: object"
      ]
     },
     "execution_count": 158,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# replace missing values with mean\n",
    "#proteins_mean = proteins_transpose[['NPC1']].mean()\n",
    "#proteins_mean"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "id": "0c51c3df",
   "metadata": {},
   "outputs": [],
   "source": [
    "#proteins_transpose['NPC1'].fillna(proteins_mean, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "id": "4ecb9171",
   "metadata": {},
   "outputs": [],
   "source": [
    "proteins_transpose.fillna(0, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "id": "e39b3c4c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "TUBA1B\n",
       "ACTG1       False\n",
       "TUBB4B      False\n",
       "HBB         False\n",
       "GAPDH       False\n",
       "HBA2        False\n",
       "            ...  \n",
       "REEP6       False\n",
       "NPC1        False\n",
       "DSCR3       False\n",
       "HAX1        False\n",
       "VKORC1L1    False\n",
       "Length: 3331, dtype: bool"
      ]
     },
     "execution_count": 162,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteins_missing_val_1 = proteins_transpose.isnull().any()\n",
    "proteins_missing_val_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "id": "14450e74",
   "metadata": {},
   "outputs": [],
   "source": [
    "proteins_transpose.to_csv(\"csv_files/proteins_new.csv\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8f7df590",
   "metadata": {},
   "source": [
    "## meta.csv\n",
    "\n",
    "Provides experimental conditions, sample IDs, age, sex, diagnosis, tissue type, and batch information for each sample."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "e842c4d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "meta = pd.read_csv('csv_files/meta.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "44ee014b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GSM</th>\n",
       "      <th>Age</th>\n",
       "      <th>Sex</th>\n",
       "      <th>Diagnosis</th>\n",
       "      <th>Tissue</th>\n",
       "      <th>Batch</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GSM3577568</td>\n",
       "      <td>93</td>\n",
       "      <td>Female</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GSM3577569</td>\n",
       "      <td>77</td>\n",
       "      <td>Male</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSM3577570</td>\n",
       "      <td>92</td>\n",
       "      <td>Male</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GSM3577571</td>\n",
       "      <td>85</td>\n",
       "      <td>Male</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GSM3577572</td>\n",
       "      <td>83</td>\n",
       "      <td>Male</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GSM3577573</td>\n",
       "      <td>85</td>\n",
       "      <td>Male</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GSM3577574</td>\n",
       "      <td>93</td>\n",
       "      <td>Female</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GSM3577575</td>\n",
       "      <td>96</td>\n",
       "      <td>Male</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GSM3577576</td>\n",
       "      <td>96</td>\n",
       "      <td>Female</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GSM3577577</td>\n",
       "      <td>88</td>\n",
       "      <td>Female</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          GSM Age     Sex            Diagnosis                 Tissue  Batch\n",
       "0  GSM3577568  93  Female  Alzheimer's disease  Brain; fusiform gyrus      1\n",
       "1  GSM3577569  77    Male  Alzheimer's disease  Brain; fusiform gyrus      1\n",
       "2  GSM3577570  92    Male  Alzheimer's disease  Brain; fusiform gyrus      1\n",
       "3  GSM3577571  85    Male  Alzheimer's disease  Brain; fusiform gyrus      1\n",
       "4  GSM3577572  83    Male  Alzheimer's disease  Brain; fusiform gyrus      1\n",
       "5  GSM3577573  85    Male  Alzheimer's disease  Brain; fusiform gyrus      1\n",
       "6  GSM3577574  93  Female  Alzheimer's disease  Brain; fusiform gyrus      1\n",
       "7  GSM3577575  96    Male  Alzheimer's disease  Brain; fusiform gyrus      1\n",
       "8  GSM3577576  96  Female  Alzheimer's disease  Brain; fusiform gyrus      1\n",
       "9  GSM3577577  88  Female  Alzheimer's disease  Brain; fusiform gyrus      1"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "4b6bb74d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# checking for missing values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "da249295",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GSM          False\n",
       "Age           True\n",
       "Sex           True\n",
       "Diagnosis    False\n",
       "Tissue       False\n",
       "Batch        False\n",
       "dtype: bool"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_missing_val = meta.isnull().any()\n",
    "meta_missing_val"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "8e1ceb2c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GSM          0\n",
       "Age          1\n",
       "Sex          1\n",
       "Diagnosis    0\n",
       "Tissue       0\n",
       "Batch        0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_missing_val = meta.isnull().sum()\n",
    "meta_missing_val"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "46b8d7e5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# encode the sex column\n",
    "le_sex = LabelEncoder()\n",
    "meta['Sex_n'] = le_sex.fit_transform(meta['Sex'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "94501203",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.9571428571428572"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# replace the missing values with mean()\n",
    "Sex_mean = meta['Sex_n'].mean()\n",
    "Sex_mean"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "a7fc9b0e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# since the mean is 0.9 and sex can't be in decimals so rounding of the value\n",
    "round(Sex_mean)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "bf440657",
   "metadata": {},
   "outputs": [],
   "source": [
    "# replace the NA valuesfor male=1, female=0\n",
    "meta['Sex_n'].fillna(round(Sex_mean), inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "d10d3a5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# replace the NA valuesfor male=1, female=0 for the string column\n",
    "meta['Sex'].fillna('Male', inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "d8e02251",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GSM</th>\n",
       "      <th>Age</th>\n",
       "      <th>Sex</th>\n",
       "      <th>Diagnosis</th>\n",
       "      <th>Tissue</th>\n",
       "      <th>Batch</th>\n",
       "      <th>Sex_n</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GSM3577568</td>\n",
       "      <td>93</td>\n",
       "      <td>Female</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GSM3577569</td>\n",
       "      <td>77</td>\n",
       "      <td>Male</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSM3577570</td>\n",
       "      <td>92</td>\n",
       "      <td>Male</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GSM3577571</td>\n",
       "      <td>85</td>\n",
       "      <td>Male</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GSM3577572</td>\n",
       "      <td>83</td>\n",
       "      <td>Male</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GSM3577573</td>\n",
       "      <td>85</td>\n",
       "      <td>Male</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GSM3577574</td>\n",
       "      <td>93</td>\n",
       "      <td>Female</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GSM3577575</td>\n",
       "      <td>96</td>\n",
       "      <td>Male</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GSM3577576</td>\n",
       "      <td>96</td>\n",
       "      <td>Female</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GSM3577577</td>\n",
       "      <td>88</td>\n",
       "      <td>Female</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          GSM Age     Sex            Diagnosis                 Tissue  Batch  \\\n",
       "0  GSM3577568  93  Female  Alzheimer's disease  Brain; fusiform gyrus      1   \n",
       "1  GSM3577569  77    Male  Alzheimer's disease  Brain; fusiform gyrus      1   \n",
       "2  GSM3577570  92    Male  Alzheimer's disease  Brain; fusiform gyrus      1   \n",
       "3  GSM3577571  85    Male  Alzheimer's disease  Brain; fusiform gyrus      1   \n",
       "4  GSM3577572  83    Male  Alzheimer's disease  Brain; fusiform gyrus      1   \n",
       "5  GSM3577573  85    Male  Alzheimer's disease  Brain; fusiform gyrus      1   \n",
       "6  GSM3577574  93  Female  Alzheimer's disease  Brain; fusiform gyrus      1   \n",
       "7  GSM3577575  96    Male  Alzheimer's disease  Brain; fusiform gyrus      1   \n",
       "8  GSM3577576  96  Female  Alzheimer's disease  Brain; fusiform gyrus      1   \n",
       "9  GSM3577577  88  Female  Alzheimer's disease  Brain; fusiform gyrus      1   \n",
       "\n",
       "   Sex_n  \n",
       "0      0  \n",
       "1      1  \n",
       "2      1  \n",
       "3      1  \n",
       "4      1  \n",
       "5      1  \n",
       "6      0  \n",
       "7      1  \n",
       "8      0  \n",
       "9      0  "
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "1f9d4586",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Convert 'Age' column to numeric, coercing errors to NaN\n",
    "meta['Age'] = pd.to_numeric(meta['Age'], errors='coerce')\n",
    "\n",
    "# Replace NaN values with 0 in 'Age' column\n",
    "meta['Age'].fillna(0, inplace=True)\n",
    "\n",
    "# Convert 'Age' column to integer\n",
    "meta['Age'] = meta['Age'].astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "d6dbee6c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "64.85714285714286"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# replace the missing values with mean()\n",
    "Age_mean = meta['Age'].mean()\n",
    "Age_mean"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "d6f4595f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# replace the NA values with mean\n",
    "meta['Age'].fillna(meta['Age'].mean(), inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "c26f6c5b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GSM</th>\n",
       "      <th>Age</th>\n",
       "      <th>Sex</th>\n",
       "      <th>Diagnosis</th>\n",
       "      <th>Tissue</th>\n",
       "      <th>Batch</th>\n",
       "      <th>Sex_n</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GSM3577568</td>\n",
       "      <td>93</td>\n",
       "      <td>Female</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GSM3577569</td>\n",
       "      <td>77</td>\n",
       "      <td>Male</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSM3577570</td>\n",
       "      <td>92</td>\n",
       "      <td>Male</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GSM3577571</td>\n",
       "      <td>85</td>\n",
       "      <td>Male</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GSM3577572</td>\n",
       "      <td>83</td>\n",
       "      <td>Male</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GSM3577573</td>\n",
       "      <td>85</td>\n",
       "      <td>Male</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GSM3577574</td>\n",
       "      <td>93</td>\n",
       "      <td>Female</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GSM3577575</td>\n",
       "      <td>96</td>\n",
       "      <td>Male</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GSM3577576</td>\n",
       "      <td>96</td>\n",
       "      <td>Female</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GSM3577577</td>\n",
       "      <td>88</td>\n",
       "      <td>Female</td>\n",
       "      <td>Alzheimer's disease</td>\n",
       "      <td>Brain; fusiform gyrus</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          GSM  Age     Sex            Diagnosis                 Tissue  Batch  \\\n",
       "0  GSM3577568   93  Female  Alzheimer's disease  Brain; fusiform gyrus      1   \n",
       "1  GSM3577569   77    Male  Alzheimer's disease  Brain; fusiform gyrus      1   \n",
       "2  GSM3577570   92    Male  Alzheimer's disease  Brain; fusiform gyrus      1   \n",
       "3  GSM3577571   85    Male  Alzheimer's disease  Brain; fusiform gyrus      1   \n",
       "4  GSM3577572   83    Male  Alzheimer's disease  Brain; fusiform gyrus      1   \n",
       "5  GSM3577573   85    Male  Alzheimer's disease  Brain; fusiform gyrus      1   \n",
       "6  GSM3577574   93  Female  Alzheimer's disease  Brain; fusiform gyrus      1   \n",
       "7  GSM3577575   96    Male  Alzheimer's disease  Brain; fusiform gyrus      1   \n",
       "8  GSM3577576   96  Female  Alzheimer's disease  Brain; fusiform gyrus      1   \n",
       "9  GSM3577577   88  Female  Alzheimer's disease  Brain; fusiform gyrus      1   \n",
       "\n",
       "   Sex_n  \n",
       "0      0  \n",
       "1      1  \n",
       "2      1  \n",
       "3      1  \n",
       "4      1  \n",
       "5      1  \n",
       "6      0  \n",
       "7      1  \n",
       "8      0  \n",
       "9      0  "
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "45f8f1c5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GSM          False\n",
       "Age          False\n",
       "Sex          False\n",
       "Diagnosis    False\n",
       "Tissue       False\n",
       "Batch        False\n",
       "Sex_n        False\n",
       "dtype: bool"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_missing_val = meta.isnull().any()\n",
    "meta_missing_val"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "b876aa2c",
   "metadata": {},
   "outputs": [],
   "source": [
    "meta.to_csv(\"csv_files/meta_new.csv\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2d6f7792",
   "metadata": {},
   "source": [
    "# combinedSTARcounts.csv\n",
    "\n",
    "Contains STAR count data from RNA seq experiments. It lists the number of unmapped reads across different samples, with each column representing a specific sample. This raw count data will require further normalization and differential expression analysis.​"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "41c65e47",
   "metadata": {},
   "outputs": [],
   "source": [
    "combinedSTAR = pd.read_csv('csv_files/combinedSTARcounts.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "a6db2224",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N_unmapped</th>\n",
       "      <th>./GSM1376252</th>\n",
       "      <th>./GSM1376253</th>\n",
       "      <th>./GSM1376254</th>\n",
       "      <th>./GSM1376255</th>\n",
       "      <th>./GSM1376256</th>\n",
       "      <th>./GSM1376257</th>\n",
       "      <th>./GSM1376258</th>\n",
       "      <th>./GSM1376259</th>\n",
       "      <th>./GSM1376260</th>\n",
       "      <th>...</th>\n",
       "      <th>./GSM4403304</th>\n",
       "      <th>./GSM4403307</th>\n",
       "      <th>./GSM4403308</th>\n",
       "      <th>./GSM4403309</th>\n",
       "      <th>./GSM4403310</th>\n",
       "      <th>./GSM4403311</th>\n",
       "      <th>./GSM4636679</th>\n",
       "      <th>./GSM4636680</th>\n",
       "      <th>./GSM4636681</th>\n",
       "      <th>./GSM4636682</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>N_multimapping</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>4947058</td>\n",
       "      <td>6332941</td>\n",
       "      <td>4128773</td>\n",
       "      <td>3722245</td>\n",
       "      <td>2963347</td>\n",
       "      <td>3127395</td>\n",
       "      <td>2856189</td>\n",
       "      <td>4964848</td>\n",
       "      <td>854938</td>\n",
       "      <td>9773683</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>N_noFeature</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>6677514</td>\n",
       "      <td>5517055</td>\n",
       "      <td>5791810</td>\n",
       "      <td>3760436</td>\n",
       "      <td>5354065</td>\n",
       "      <td>4291684</td>\n",
       "      <td>5515906</td>\n",
       "      <td>3673654</td>\n",
       "      <td>853466</td>\n",
       "      <td>3756792</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>N_ambiguous</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>125207</td>\n",
       "      <td>118670</td>\n",
       "      <td>141932</td>\n",
       "      <td>161794</td>\n",
       "      <td>162896</td>\n",
       "      <td>179798</td>\n",
       "      <td>118270</td>\n",
       "      <td>56726</td>\n",
       "      <td>16497</td>\n",
       "      <td>103998</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>WASH7P</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>62</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MIR6859-3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MIR6859-2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MIR6859-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MIR6859-4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MIR1302-2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 768 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       N_unmapped  ./GSM1376252  ./GSM1376253  ./GSM1376254  ./GSM1376255  \\\n",
       "0  N_multimapping             0             0             0             0   \n",
       "1     N_noFeature             0             0             0             0   \n",
       "2     N_ambiguous             0             0             0             0   \n",
       "3         DDX11L1             0             0             0             0   \n",
       "4          WASH7P             0             0             0             0   \n",
       "5       MIR6859-3             0             0             0             0   \n",
       "6       MIR6859-2             0             0             0             0   \n",
       "7       MIR6859-1             0             0             0             0   \n",
       "8       MIR6859-4             0             0             0             0   \n",
       "9       MIR1302-2             0             0             0             0   \n",
       "\n",
       "   ./GSM1376256  ./GSM1376257  ./GSM1376258  ./GSM1376259  ./GSM1376260  ...  \\\n",
       "0             0             0             0             0             0  ...   \n",
       "1             0             0             0             0             0  ...   \n",
       "2             0             0             0             0             0  ...   \n",
       "3             0             0             0             0             0  ...   \n",
       "4             0             0             0             0             0  ...   \n",
       "5             0             0             0             0             0  ...   \n",
       "6             0             0             0             0             0  ...   \n",
       "7             0             0             0             0             0  ...   \n",
       "8             0             0             0             0             0  ...   \n",
       "9             0             0             0             0             0  ...   \n",
       "\n",
       "   ./GSM4403304  ./GSM4403307  ./GSM4403308  ./GSM4403309  ./GSM4403310  \\\n",
       "0       4947058       6332941       4128773       3722245       2963347   \n",
       "1       6677514       5517055       5791810       3760436       5354065   \n",
       "2        125207        118670        141932        161794        162896   \n",
       "3             0             0             0             0             0   \n",
       "4             0            62             0            23             0   \n",
       "5             0             0             0             0             0   \n",
       "6             0             0             0             0             0   \n",
       "7             0             0             0             0             0   \n",
       "8             0             0             0             0             0   \n",
       "9             0             0             0             0             0   \n",
       "\n",
       "   ./GSM4403311  ./GSM4636679  ./GSM4636680  ./GSM4636681  ./GSM4636682  \n",
       "0       3127395       2856189       4964848        854938       9773683  \n",
       "1       4291684       5515906       3673654        853466       3756792  \n",
       "2        179798        118270         56726         16497        103998  \n",
       "3             0             0             0             0             0  \n",
       "4             0             0             0             0             0  \n",
       "5             0             0             0             0             0  \n",
       "6             0             0             0             0             0  \n",
       "7             0             0             0             0             0  \n",
       "8             0             0             0             0             0  \n",
       "9             0             0             0             0             0  \n",
       "\n",
       "[10 rows x 768 columns]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combinedSTAR.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "290542ba",
   "metadata": {},
   "outputs": [],
   "source": [
    "# checking for missing values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "89567fe5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "N_unmapped      False\n",
       "./GSM1376252    False\n",
       "./GSM1376253    False\n",
       "./GSM1376254    False\n",
       "./GSM1376255    False\n",
       "                ...  \n",
       "./GSM4403311    False\n",
       "./GSM4636679    False\n",
       "./GSM4636680    False\n",
       "./GSM4636681    False\n",
       "./GSM4636682    False\n",
       "Length: 768, dtype: bool"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combinedSTAR_missing_val = combinedSTAR.isnull().any()\n",
    "combinedSTAR_missing_val"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "6301294b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "N_unmapped      0\n",
       "./GSM1376252    0\n",
       "./GSM1376253    0\n",
       "./GSM1376254    0\n",
       "./GSM1376255    0\n",
       "               ..\n",
       "./GSM4403311    0\n",
       "./GSM4636679    0\n",
       "./GSM4636680    0\n",
       "./GSM4636681    0\n",
       "./GSM4636682    0\n",
       "Length: 768, dtype: int64"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combinedSTAR_missing_val = combinedSTAR.isnull().sum()\n",
    "combinedSTAR_missing_val"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "ef86b140",
   "metadata": {},
   "outputs": [],
   "source": [
    "combinedSTAR.to_csv(\"csv_files/combinedSTAR_new.csv\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "178c695c",
   "metadata": {},
   "source": [
    "# Microarray.csv\n",
    "\n",
    "Contains Microarray expression data for cross-validation and comparative analysis with RNA-seq data.​"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "1c4a47a9",
   "metadata": {},
   "outputs": [],
   "source": [
    "microarray = pd.read_csv('csv_files/Microarray.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "0d79d3f9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>V1</th>\n",
       "      <th>V2</th>\n",
       "      <th>V3</th>\n",
       "      <th>V4</th>\n",
       "      <th>V5</th>\n",
       "      <th>V6</th>\n",
       "      <th>V7</th>\n",
       "      <th>V8</th>\n",
       "      <th>V9</th>\n",
       "      <th>V10</th>\n",
       "      <th>...</th>\n",
       "      <th>V2037</th>\n",
       "      <th>V2038</th>\n",
       "      <th>V2039</th>\n",
       "      <th>V2040</th>\n",
       "      <th>V2041</th>\n",
       "      <th>V2042</th>\n",
       "      <th>V2043</th>\n",
       "      <th>V2044</th>\n",
       "      <th>V2045</th>\n",
       "      <th>V2046</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>5.582546</td>\n",
       "      <td>5.316083</td>\n",
       "      <td>5.639426</td>\n",
       "      <td>5.531861</td>\n",
       "      <td>5.492184</td>\n",
       "      <td>5.619112</td>\n",
       "      <td>5.424081</td>\n",
       "      <td>5.370087</td>\n",
       "      <td>5.597164</td>\n",
       "      <td>...</td>\n",
       "      <td>5.503977</td>\n",
       "      <td>5.435669</td>\n",
       "      <td>5.291581</td>\n",
       "      <td>5.293762</td>\n",
       "      <td>5.545232</td>\n",
       "      <td>5.413534</td>\n",
       "      <td>5.326086</td>\n",
       "      <td>5.315767</td>\n",
       "      <td>5.220495</td>\n",
       "      <td>5.657885</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A4GALT</td>\n",
       "      <td>4.588665</td>\n",
       "      <td>4.567148</td>\n",
       "      <td>4.336331</td>\n",
       "      <td>4.464754</td>\n",
       "      <td>4.454753</td>\n",
       "      <td>4.382402</td>\n",
       "      <td>4.352616</td>\n",
       "      <td>4.478181</td>\n",
       "      <td>4.816772</td>\n",
       "      <td>...</td>\n",
       "      <td>4.559906</td>\n",
       "      <td>4.542629</td>\n",
       "      <td>4.410823</td>\n",
       "      <td>4.337349</td>\n",
       "      <td>4.479923</td>\n",
       "      <td>4.414422</td>\n",
       "      <td>4.521428</td>\n",
       "      <td>4.608279</td>\n",
       "      <td>4.484420</td>\n",
       "      <td>4.604231</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A4GNT</td>\n",
       "      <td>4.729963</td>\n",
       "      <td>4.607885</td>\n",
       "      <td>4.419786</td>\n",
       "      <td>4.691284</td>\n",
       "      <td>4.967895</td>\n",
       "      <td>4.542279</td>\n",
       "      <td>4.597191</td>\n",
       "      <td>4.618374</td>\n",
       "      <td>5.458812</td>\n",
       "      <td>...</td>\n",
       "      <td>4.604645</td>\n",
       "      <td>4.593086</td>\n",
       "      <td>4.841435</td>\n",
       "      <td>4.994085</td>\n",
       "      <td>4.337664</td>\n",
       "      <td>4.805862</td>\n",
       "      <td>4.600341</td>\n",
       "      <td>4.278393</td>\n",
       "      <td>4.554369</td>\n",
       "      <td>4.418984</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>5.201235</td>\n",
       "      <td>4.771399</td>\n",
       "      <td>5.024912</td>\n",
       "      <td>5.222125</td>\n",
       "      <td>5.172302</td>\n",
       "      <td>4.980001</td>\n",
       "      <td>5.138890</td>\n",
       "      <td>5.060142</td>\n",
       "      <td>5.041990</td>\n",
       "      <td>...</td>\n",
       "      <td>5.044219</td>\n",
       "      <td>4.994378</td>\n",
       "      <td>4.879115</td>\n",
       "      <td>5.019100</td>\n",
       "      <td>4.932031</td>\n",
       "      <td>4.966196</td>\n",
       "      <td>5.009232</td>\n",
       "      <td>4.928180</td>\n",
       "      <td>4.983588</td>\n",
       "      <td>5.001185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AACS</td>\n",
       "      <td>5.516770</td>\n",
       "      <td>5.090253</td>\n",
       "      <td>5.129032</td>\n",
       "      <td>5.502365</td>\n",
       "      <td>5.142271</td>\n",
       "      <td>5.624660</td>\n",
       "      <td>5.486132</td>\n",
       "      <td>5.519051</td>\n",
       "      <td>5.317086</td>\n",
       "      <td>...</td>\n",
       "      <td>5.126910</td>\n",
       "      <td>4.503967</td>\n",
       "      <td>5.678005</td>\n",
       "      <td>5.670121</td>\n",
       "      <td>5.413771</td>\n",
       "      <td>5.265686</td>\n",
       "      <td>5.389198</td>\n",
       "      <td>5.054624</td>\n",
       "      <td>4.976820</td>\n",
       "      <td>5.459347</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>AADAT</td>\n",
       "      <td>4.321050</td>\n",
       "      <td>4.223453</td>\n",
       "      <td>4.564527</td>\n",
       "      <td>4.366150</td>\n",
       "      <td>4.431567</td>\n",
       "      <td>4.151446</td>\n",
       "      <td>4.441678</td>\n",
       "      <td>4.442235</td>\n",
       "      <td>4.316715</td>\n",
       "      <td>...</td>\n",
       "      <td>4.308273</td>\n",
       "      <td>4.382805</td>\n",
       "      <td>4.222056</td>\n",
       "      <td>4.282696</td>\n",
       "      <td>4.307498</td>\n",
       "      <td>4.231064</td>\n",
       "      <td>4.393983</td>\n",
       "      <td>4.120549</td>\n",
       "      <td>4.181310</td>\n",
       "      <td>4.310877</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>AAK1</td>\n",
       "      <td>5.403211</td>\n",
       "      <td>5.355780</td>\n",
       "      <td>5.159841</td>\n",
       "      <td>5.443067</td>\n",
       "      <td>5.223076</td>\n",
       "      <td>5.635608</td>\n",
       "      <td>5.593506</td>\n",
       "      <td>5.427302</td>\n",
       "      <td>5.406536</td>\n",
       "      <td>...</td>\n",
       "      <td>5.555753</td>\n",
       "      <td>5.738224</td>\n",
       "      <td>5.531037</td>\n",
       "      <td>5.130048</td>\n",
       "      <td>5.610595</td>\n",
       "      <td>5.472773</td>\n",
       "      <td>5.670954</td>\n",
       "      <td>5.781740</td>\n",
       "      <td>5.749025</td>\n",
       "      <td>5.588248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>AAMP</td>\n",
       "      <td>4.682736</td>\n",
       "      <td>4.615632</td>\n",
       "      <td>4.621200</td>\n",
       "      <td>4.537199</td>\n",
       "      <td>4.595105</td>\n",
       "      <td>4.687976</td>\n",
       "      <td>4.806862</td>\n",
       "      <td>4.681141</td>\n",
       "      <td>4.454123</td>\n",
       "      <td>...</td>\n",
       "      <td>4.776131</td>\n",
       "      <td>4.990350</td>\n",
       "      <td>4.738362</td>\n",
       "      <td>4.744300</td>\n",
       "      <td>4.696910</td>\n",
       "      <td>4.777630</td>\n",
       "      <td>4.629461</td>\n",
       "      <td>4.830475</td>\n",
       "      <td>4.762774</td>\n",
       "      <td>4.685090</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>AARS</td>\n",
       "      <td>5.779412</td>\n",
       "      <td>5.892408</td>\n",
       "      <td>5.458466</td>\n",
       "      <td>5.836839</td>\n",
       "      <td>5.531098</td>\n",
       "      <td>5.783979</td>\n",
       "      <td>5.766949</td>\n",
       "      <td>5.546404</td>\n",
       "      <td>5.625244</td>\n",
       "      <td>...</td>\n",
       "      <td>5.542206</td>\n",
       "      <td>5.485895</td>\n",
       "      <td>5.510172</td>\n",
       "      <td>5.619324</td>\n",
       "      <td>5.566223</td>\n",
       "      <td>5.592551</td>\n",
       "      <td>5.572670</td>\n",
       "      <td>5.709698</td>\n",
       "      <td>5.518494</td>\n",
       "      <td>5.450246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>AASDH</td>\n",
       "      <td>4.476559</td>\n",
       "      <td>4.677460</td>\n",
       "      <td>4.310792</td>\n",
       "      <td>4.407091</td>\n",
       "      <td>4.347562</td>\n",
       "      <td>4.303371</td>\n",
       "      <td>4.446790</td>\n",
       "      <td>4.317594</td>\n",
       "      <td>4.316065</td>\n",
       "      <td>...</td>\n",
       "      <td>4.620338</td>\n",
       "      <td>4.314468</td>\n",
       "      <td>4.265021</td>\n",
       "      <td>4.371578</td>\n",
       "      <td>4.237609</td>\n",
       "      <td>4.511817</td>\n",
       "      <td>4.429191</td>\n",
       "      <td>4.552967</td>\n",
       "      <td>4.488417</td>\n",
       "      <td>4.312414</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 2046 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       V1        V2        V3        V4        V5        V6        V7  \\\n",
       "0    A1BG  5.582546  5.316083  5.639426  5.531861  5.492184  5.619112   \n",
       "1  A4GALT  4.588665  4.567148  4.336331  4.464754  4.454753  4.382402   \n",
       "2   A4GNT  4.729963  4.607885  4.419786  4.691284  4.967895  4.542279   \n",
       "3    AAAS  5.201235  4.771399  5.024912  5.222125  5.172302  4.980001   \n",
       "4    AACS  5.516770  5.090253  5.129032  5.502365  5.142271  5.624660   \n",
       "5   AADAT  4.321050  4.223453  4.564527  4.366150  4.431567  4.151446   \n",
       "6    AAK1  5.403211  5.355780  5.159841  5.443067  5.223076  5.635608   \n",
       "7    AAMP  4.682736  4.615632  4.621200  4.537199  4.595105  4.687976   \n",
       "8    AARS  5.779412  5.892408  5.458466  5.836839  5.531098  5.783979   \n",
       "9   AASDH  4.476559  4.677460  4.310792  4.407091  4.347562  4.303371   \n",
       "\n",
       "         V8        V9       V10  ...     V2037     V2038     V2039     V2040  \\\n",
       "0  5.424081  5.370087  5.597164  ...  5.503977  5.435669  5.291581  5.293762   \n",
       "1  4.352616  4.478181  4.816772  ...  4.559906  4.542629  4.410823  4.337349   \n",
       "2  4.597191  4.618374  5.458812  ...  4.604645  4.593086  4.841435  4.994085   \n",
       "3  5.138890  5.060142  5.041990  ...  5.044219  4.994378  4.879115  5.019100   \n",
       "4  5.486132  5.519051  5.317086  ...  5.126910  4.503967  5.678005  5.670121   \n",
       "5  4.441678  4.442235  4.316715  ...  4.308273  4.382805  4.222056  4.282696   \n",
       "6  5.593506  5.427302  5.406536  ...  5.555753  5.738224  5.531037  5.130048   \n",
       "7  4.806862  4.681141  4.454123  ...  4.776131  4.990350  4.738362  4.744300   \n",
       "8  5.766949  5.546404  5.625244  ...  5.542206  5.485895  5.510172  5.619324   \n",
       "9  4.446790  4.317594  4.316065  ...  4.620338  4.314468  4.265021  4.371578   \n",
       "\n",
       "      V2041     V2042     V2043     V2044     V2045     V2046  \n",
       "0  5.545232  5.413534  5.326086  5.315767  5.220495  5.657885  \n",
       "1  4.479923  4.414422  4.521428  4.608279  4.484420  4.604231  \n",
       "2  4.337664  4.805862  4.600341  4.278393  4.554369  4.418984  \n",
       "3  4.932031  4.966196  5.009232  4.928180  4.983588  5.001185  \n",
       "4  5.413771  5.265686  5.389198  5.054624  4.976820  5.459347  \n",
       "5  4.307498  4.231064  4.393983  4.120549  4.181310  4.310877  \n",
       "6  5.610595  5.472773  5.670954  5.781740  5.749025  5.588248  \n",
       "7  4.696910  4.777630  4.629461  4.830475  4.762774  4.685090  \n",
       "8  5.566223  5.592551  5.572670  5.709698  5.518494  5.450246  \n",
       "9  4.237609  4.511817  4.429191  4.552967  4.488417  4.312414  \n",
       "\n",
       "[10 rows x 2046 columns]"
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "microarray.head(10) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "b3ff7d76",
   "metadata": {},
   "outputs": [],
   "source": [
    "# checking for missing values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "27d186e8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "V1       False\n",
       "V2       False\n",
       "V3       False\n",
       "V4       False\n",
       "V5       False\n",
       "         ...  \n",
       "V2042    False\n",
       "V2043    False\n",
       "V2044    False\n",
       "V2045    False\n",
       "V2046    False\n",
       "Length: 2046, dtype: bool"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "microarray_missing_val = microarray.isnull().any()\n",
    "microarray_missing_val"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "01a8215f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "V1       0\n",
       "V2       0\n",
       "V3       0\n",
       "V4       0\n",
       "V5       0\n",
       "        ..\n",
       "V2042    0\n",
       "V2043    0\n",
       "V2044    0\n",
       "V2045    0\n",
       "V2046    0\n",
       "Length: 2046, dtype: int64"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "microarray_missing_val = microarray.isnull().sum()\n",
    "microarray_missing_val"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "f434e6f0",
   "metadata": {},
   "outputs": [],
   "source": [
    "microarray.to_csv(\"csv_files/microarray_new.csv\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6c4a8914",
   "metadata": {},
   "source": [
    "# miRNA\n",
    "\n",
    "File includes microRNA expression profiles across different samples. Each column represents a specific sample, and this data will aid in identifying post-transcriptional regulatory mechanisms in Alzheimer's disease.​"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "84c784d0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>miRNA_IDS</th>\n",
       "      <th>GSM3403761</th>\n",
       "      <th>GSM3403762</th>\n",
       "      <th>GSM3403763</th>\n",
       "      <th>GSM3403764</th>\n",
       "      <th>GSM3403765</th>\n",
       "      <th>GSM3403766</th>\n",
       "      <th>GSM3403767</th>\n",
       "      <th>GSM3403768</th>\n",
       "      <th>GSM3403769</th>\n",
       "      <th>...</th>\n",
       "      <th>GSM3405261</th>\n",
       "      <th>GSM3405262</th>\n",
       "      <th>GSM3405263</th>\n",
       "      <th>GSM3405264</th>\n",
       "      <th>GSM3405265</th>\n",
       "      <th>GSM3405266</th>\n",
       "      <th>GSM3405267</th>\n",
       "      <th>GSM3405268</th>\n",
       "      <th>GSM3405269</th>\n",
       "      <th>GSM3405270</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>2.307579</td>\n",
       "      <td>1.503044</td>\n",
       "      <td>1.549877</td>\n",
       "      <td>1.560269</td>\n",
       "      <td>3.179096</td>\n",
       "      <td>4.319297</td>\n",
       "      <td>0.575922</td>\n",
       "      <td>-0.180237</td>\n",
       "      <td>1.491916</td>\n",
       "      <td>...</td>\n",
       "      <td>4.151464</td>\n",
       "      <td>1.547006</td>\n",
       "      <td>2.31427</td>\n",
       "      <td>1.183973</td>\n",
       "      <td>2.441098</td>\n",
       "      <td>3.093713</td>\n",
       "      <td>2.854996</td>\n",
       "      <td>3.629315</td>\n",
       "      <td>2.138062</td>\n",
       "      <td>1.896355</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0000063</td>\n",
       "      <td>2.307579</td>\n",
       "      <td>2.505380</td>\n",
       "      <td>1.983125</td>\n",
       "      <td>1.560269</td>\n",
       "      <td>3.302472</td>\n",
       "      <td>4.319297</td>\n",
       "      <td>1.964171</td>\n",
       "      <td>0.697365</td>\n",
       "      <td>1.491916</td>\n",
       "      <td>...</td>\n",
       "      <td>4.151464</td>\n",
       "      <td>1.547006</td>\n",
       "      <td>2.31427</td>\n",
       "      <td>1.183973</td>\n",
       "      <td>1.571320</td>\n",
       "      <td>3.517899</td>\n",
       "      <td>2.854996</td>\n",
       "      <td>3.013971</td>\n",
       "      <td>2.138062</td>\n",
       "      <td>1.664736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0000064</td>\n",
       "      <td>2.307579</td>\n",
       "      <td>1.503044</td>\n",
       "      <td>1.549877</td>\n",
       "      <td>1.560269</td>\n",
       "      <td>3.179096</td>\n",
       "      <td>4.319297</td>\n",
       "      <td>0.575922</td>\n",
       "      <td>-0.180237</td>\n",
       "      <td>1.491916</td>\n",
       "      <td>...</td>\n",
       "      <td>4.151464</td>\n",
       "      <td>1.547006</td>\n",
       "      <td>2.31427</td>\n",
       "      <td>2.286674</td>\n",
       "      <td>1.571320</td>\n",
       "      <td>3.093713</td>\n",
       "      <td>2.854996</td>\n",
       "      <td>3.013971</td>\n",
       "      <td>2.138062</td>\n",
       "      <td>3.253439</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0000065</td>\n",
       "      <td>2.307579</td>\n",
       "      <td>1.503044</td>\n",
       "      <td>1.549877</td>\n",
       "      <td>1.560269</td>\n",
       "      <td>3.179096</td>\n",
       "      <td>4.319297</td>\n",
       "      <td>0.575922</td>\n",
       "      <td>-0.180237</td>\n",
       "      <td>1.491916</td>\n",
       "      <td>...</td>\n",
       "      <td>4.151464</td>\n",
       "      <td>1.547006</td>\n",
       "      <td>2.31427</td>\n",
       "      <td>1.183973</td>\n",
       "      <td>1.571320</td>\n",
       "      <td>3.093713</td>\n",
       "      <td>2.854996</td>\n",
       "      <td>3.013971</td>\n",
       "      <td>2.138062</td>\n",
       "      <td>4.077636</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0000066</td>\n",
       "      <td>2.307579</td>\n",
       "      <td>1.503044</td>\n",
       "      <td>1.549877</td>\n",
       "      <td>2.232974</td>\n",
       "      <td>4.793470</td>\n",
       "      <td>4.319297</td>\n",
       "      <td>0.575922</td>\n",
       "      <td>-0.180237</td>\n",
       "      <td>1.491916</td>\n",
       "      <td>...</td>\n",
       "      <td>4.151464</td>\n",
       "      <td>1.547006</td>\n",
       "      <td>2.31427</td>\n",
       "      <td>2.777865</td>\n",
       "      <td>1.571320</td>\n",
       "      <td>3.093713</td>\n",
       "      <td>2.854996</td>\n",
       "      <td>3.013971</td>\n",
       "      <td>2.138062</td>\n",
       "      <td>1.664736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MIMAT0000067</td>\n",
       "      <td>2.307579</td>\n",
       "      <td>1.503044</td>\n",
       "      <td>1.549877</td>\n",
       "      <td>1.560269</td>\n",
       "      <td>3.179096</td>\n",
       "      <td>4.319297</td>\n",
       "      <td>0.575922</td>\n",
       "      <td>-0.180237</td>\n",
       "      <td>1.491916</td>\n",
       "      <td>...</td>\n",
       "      <td>4.151464</td>\n",
       "      <td>1.547006</td>\n",
       "      <td>2.31427</td>\n",
       "      <td>1.183973</td>\n",
       "      <td>1.571320</td>\n",
       "      <td>3.093713</td>\n",
       "      <td>2.854996</td>\n",
       "      <td>3.013971</td>\n",
       "      <td>2.138062</td>\n",
       "      <td>1.664736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MIMAT0000068</td>\n",
       "      <td>2.307579</td>\n",
       "      <td>1.503044</td>\n",
       "      <td>1.549877</td>\n",
       "      <td>1.560269</td>\n",
       "      <td>3.179096</td>\n",
       "      <td>4.319297</td>\n",
       "      <td>0.575922</td>\n",
       "      <td>-0.180237</td>\n",
       "      <td>1.491916</td>\n",
       "      <td>...</td>\n",
       "      <td>4.971346</td>\n",
       "      <td>1.547006</td>\n",
       "      <td>2.31427</td>\n",
       "      <td>2.593765</td>\n",
       "      <td>1.571320</td>\n",
       "      <td>3.093713</td>\n",
       "      <td>2.854996</td>\n",
       "      <td>3.013971</td>\n",
       "      <td>2.138062</td>\n",
       "      <td>1.664736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MIMAT0000069</td>\n",
       "      <td>2.307579</td>\n",
       "      <td>1.503044</td>\n",
       "      <td>1.549877</td>\n",
       "      <td>1.560269</td>\n",
       "      <td>3.179096</td>\n",
       "      <td>4.319297</td>\n",
       "      <td>0.575922</td>\n",
       "      <td>-0.180237</td>\n",
       "      <td>1.491916</td>\n",
       "      <td>...</td>\n",
       "      <td>4.151464</td>\n",
       "      <td>1.547006</td>\n",
       "      <td>2.31427</td>\n",
       "      <td>1.183973</td>\n",
       "      <td>1.571320</td>\n",
       "      <td>3.093713</td>\n",
       "      <td>2.854996</td>\n",
       "      <td>3.013971</td>\n",
       "      <td>2.138062</td>\n",
       "      <td>1.664736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MIMAT0000070</td>\n",
       "      <td>2.307579</td>\n",
       "      <td>1.503044</td>\n",
       "      <td>1.549877</td>\n",
       "      <td>1.560269</td>\n",
       "      <td>3.179096</td>\n",
       "      <td>4.319297</td>\n",
       "      <td>0.575922</td>\n",
       "      <td>-0.180237</td>\n",
       "      <td>1.491916</td>\n",
       "      <td>...</td>\n",
       "      <td>4.151464</td>\n",
       "      <td>3.397922</td>\n",
       "      <td>2.31427</td>\n",
       "      <td>1.183973</td>\n",
       "      <td>1.821418</td>\n",
       "      <td>3.093713</td>\n",
       "      <td>2.854996</td>\n",
       "      <td>3.013971</td>\n",
       "      <td>2.138062</td>\n",
       "      <td>1.664736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MIMAT0000071</td>\n",
       "      <td>2.843409</td>\n",
       "      <td>3.349936</td>\n",
       "      <td>3.081569</td>\n",
       "      <td>1.890881</td>\n",
       "      <td>3.650599</td>\n",
       "      <td>4.319297</td>\n",
       "      <td>0.575922</td>\n",
       "      <td>0.556748</td>\n",
       "      <td>3.346453</td>\n",
       "      <td>...</td>\n",
       "      <td>4.151464</td>\n",
       "      <td>3.541742</td>\n",
       "      <td>2.31427</td>\n",
       "      <td>1.183973</td>\n",
       "      <td>1.571320</td>\n",
       "      <td>3.093713</td>\n",
       "      <td>2.854996</td>\n",
       "      <td>3.013971</td>\n",
       "      <td>2.138062</td>\n",
       "      <td>1.664736</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 1310 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      miRNA_IDS  GSM3403761  GSM3403762  GSM3403763  GSM3403764  GSM3403765  \\\n",
       "0  MIMAT0000062    2.307579    1.503044    1.549877    1.560269    3.179096   \n",
       "1  MIMAT0000063    2.307579    2.505380    1.983125    1.560269    3.302472   \n",
       "2  MIMAT0000064    2.307579    1.503044    1.549877    1.560269    3.179096   \n",
       "3  MIMAT0000065    2.307579    1.503044    1.549877    1.560269    3.179096   \n",
       "4  MIMAT0000066    2.307579    1.503044    1.549877    2.232974    4.793470   \n",
       "5  MIMAT0000067    2.307579    1.503044    1.549877    1.560269    3.179096   \n",
       "6  MIMAT0000068    2.307579    1.503044    1.549877    1.560269    3.179096   \n",
       "7  MIMAT0000069    2.307579    1.503044    1.549877    1.560269    3.179096   \n",
       "8  MIMAT0000070    2.307579    1.503044    1.549877    1.560269    3.179096   \n",
       "9  MIMAT0000071    2.843409    3.349936    3.081569    1.890881    3.650599   \n",
       "\n",
       "   GSM3403766  GSM3403767  GSM3403768  GSM3403769  ...  GSM3405261  \\\n",
       "0    4.319297    0.575922   -0.180237    1.491916  ...    4.151464   \n",
       "1    4.319297    1.964171    0.697365    1.491916  ...    4.151464   \n",
       "2    4.319297    0.575922   -0.180237    1.491916  ...    4.151464   \n",
       "3    4.319297    0.575922   -0.180237    1.491916  ...    4.151464   \n",
       "4    4.319297    0.575922   -0.180237    1.491916  ...    4.151464   \n",
       "5    4.319297    0.575922   -0.180237    1.491916  ...    4.151464   \n",
       "6    4.319297    0.575922   -0.180237    1.491916  ...    4.971346   \n",
       "7    4.319297    0.575922   -0.180237    1.491916  ...    4.151464   \n",
       "8    4.319297    0.575922   -0.180237    1.491916  ...    4.151464   \n",
       "9    4.319297    0.575922    0.556748    3.346453  ...    4.151464   \n",
       "\n",
       "   GSM3405262  GSM3405263  GSM3405264  GSM3405265  GSM3405266  GSM3405267  \\\n",
       "0    1.547006     2.31427    1.183973    2.441098    3.093713    2.854996   \n",
       "1    1.547006     2.31427    1.183973    1.571320    3.517899    2.854996   \n",
       "2    1.547006     2.31427    2.286674    1.571320    3.093713    2.854996   \n",
       "3    1.547006     2.31427    1.183973    1.571320    3.093713    2.854996   \n",
       "4    1.547006     2.31427    2.777865    1.571320    3.093713    2.854996   \n",
       "5    1.547006     2.31427    1.183973    1.571320    3.093713    2.854996   \n",
       "6    1.547006     2.31427    2.593765    1.571320    3.093713    2.854996   \n",
       "7    1.547006     2.31427    1.183973    1.571320    3.093713    2.854996   \n",
       "8    3.397922     2.31427    1.183973    1.821418    3.093713    2.854996   \n",
       "9    3.541742     2.31427    1.183973    1.571320    3.093713    2.854996   \n",
       "\n",
       "   GSM3405268  GSM3405269  GSM3405270  \n",
       "0    3.629315    2.138062    1.896355  \n",
       "1    3.013971    2.138062    1.664736  \n",
       "2    3.013971    2.138062    3.253439  \n",
       "3    3.013971    2.138062    4.077636  \n",
       "4    3.013971    2.138062    1.664736  \n",
       "5    3.013971    2.138062    1.664736  \n",
       "6    3.013971    2.138062    1.664736  \n",
       "7    3.013971    2.138062    1.664736  \n",
       "8    3.013971    2.138062    1.664736  \n",
       "9    3.013971    2.138062    1.664736  \n",
       "\n",
       "[10 rows x 1310 columns]"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "miRNA = pd.read_csv('csv_files/miRNA.csv')\n",
    "miRNA.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "b1c30556",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "miRNA_IDS     False\n",
       "GSM3403761    False\n",
       "GSM3403762    False\n",
       "GSM3403763    False\n",
       "GSM3403764    False\n",
       "              ...  \n",
       "GSM3405266    False\n",
       "GSM3405267    False\n",
       "GSM3405268    False\n",
       "GSM3405269    False\n",
       "GSM3405270    False\n",
       "Length: 1310, dtype: bool"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "miRNA_missing_val = miRNA.isnull().any()\n",
    "miRNA_missing_val"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "695e69e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "miRNA.to_csv(\"csv_files/miRNA_new.csv\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8297d93d",
   "metadata": {},
   "source": [
    "# AlzheimerTermsNonRedundant.txt  \n",
    "\n",
    "File is a curated list of pathways associated with Alzheimer's disease, including relevant genes and URLs for further reference. This data is essential for pathway enrichment analysis and understanding the specific biological processes involved in Alzheimer's pathology.​"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5055206e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# no prepsrocessing needs to be done on this file"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
